test/test_other.py

```diff
@@ -3884,6 +3884,27 @@ def test_js_lib_quoted_key(self):
 
     self.do_runf(test_file('hello_world.c'), 'hello, world!', emcc_args=['--js-library', 'lib.js'])
 
+  def test_js_lib_proxying(self):
+    # See https://github.com/emscripten-core/emscripten/issues/20264
+    create_file('lib.js', r'''
+addToLibrary({
+  foo__proxy: 'sync',
+  foo: () => setTimeout(() => {
+    console.log('should not see this');
```

Comment:
Why should this not be seen? `foo` is called.

Reply:
But the timeout here is 1000 seconds.. and the problem exits before we even return to the event loop.

The bug here was that the outer `setTimeout(() => {` was being stripped by the jsifier code.. resulting in the `console.log` running directly rather than on a timeout.

Reply:
It's 1000 **ms**, not seconds, so I can see a VM stall cause it to print.

I guess I'd be ok with `1000*60*15` (15 minutes) or such, as that is the timeout of CI anyhow...

Reply:
(lgtm with that, or more)

Reply:
Ah.. well in that case maybe we should make the value zero, assuming the test passes.  

I removed the timeout completely and test still passes.  Also added a comment.

Reply:
(Was that change pushed? I still see a timeout, and don't see a comment about the timeout duration of 1000)

Reply:
Oops, I'm switching to using my own emscripten branch as `origin` and it can be confusing.